internal/operands/vm-console-proxy/reconcile.go

```diff
@@ -0,0 +1,163 @@
+package vm_console_proxy
+
+import (
+	"fmt"
+
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+	vm_console_proxy_bundle "kubevirt.io/ssp-operator/internal/vm-console-proxy-bundle"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	operandName      = "vm-console-proxy"
+	operandComponent = "vm-console-proxy"
+)
+
+type vmConsoleProxy struct {
+	ServiceAccounts     []core.ServiceAccount
+	ClusterRoles        []rbac.ClusterRole
+	ClusterRoleBindings []rbac.ClusterRoleBinding
+	Services            []core.Service
+	Deployments         []apps.Deployment
+	ConfigMaps          []core.ConfigMap
+}
+
+var _ operands.Operand = &vmConsoleProxy{}
+
+func New(bundle *vm_console_proxy_bundle.Bundle) *vmConsoleProxy {
+	return &vmConsoleProxy{
+		ServiceAccounts:     bundle.ServiceAccounts,
+		ClusterRoles:        bundle.ClusterRoles,
+		ClusterRoleBindings: bundle.ClusterRoleBindings,
+		Services:            bundle.Services,
+		Deployments:         bundle.Deployments,
+		ConfigMaps:          bundle.ConfigMaps,
+	}
+}
+
+func (t *vmConsoleProxy) Name() string {
+	return operandName
+}
+
+func (t *vmConsoleProxy) WatchTypes() []operands.WatchType {
+	return nil
+}
+
+func (t *vmConsoleProxy) WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &core.ServiceAccount{}},
+		{Object: &rbac.ClusterRole{}},
+		{Object: &rbac.ClusterRoleBinding{}},
+		{Object: &core.Service{}},
+		{Object: &apps.Deployment{}},
+		{Object: &core.ConfigMap{}},
```

Comment:
All the objects, except `ClusterRole` and `ClusterRoleBinding`, should be moved to the above function `WatchTypes()`. The difference is that, cluster types can be in a different namespace than `SSP` CR and need code in `Cleanup()` function to remove them.

Reply:
Gotcha, thanks :)